pkg/pgmodel/pgx.go

```diff
@@ -322,12 +364,68 @@ const (
 	flushTimeout = 500 * time.Millisecond
 )
 
-func runInserterRoutine(conn pgxConn, input chan insertDataRequest) {
+func getMetricTableName(conn pgxConn, metric string) (string, bool, error) {
+	res, err := conn.Query(
+		context.Background(),
+		getCreateMetricsTableWithNewSQL,
+		metric,
+	)
+
+	if err != nil {
+		return "", true, err
+	}
+
+	var tableName string
+	var possiblyNew bool
+	defer res.Close()
+	if !res.Next() {
+		return "", true, errMissingTableName
+	}
+
+	if err := res.Scan(&tableName, &possiblyNew); err != nil {
+		return "", true, err
+	}
+
+	return tableName, possiblyNew, nil
+}
+
+func runInserterRoutine(conn pgxConn, input chan insertDataRequest, metricName string, completeMetricCreationSignal chan bool, errChan chan error, metricTableNames MetricCache) {
+	tableName, err := metricTableNames.Get(metricName)
+	if err == ErrEntryNotFound {
+		var possiblyNew bool
+		tableName, possiblyNew, err = getMetricTableName(conn, metricName)
+		if err != nil {
+			select {
+			case errChan <- err:
```

Comment:
In this case, when we we don't have a `tableName` we should exit the inserter routine; we won't be able to insert anyway.